src/main/java/org/photonvision/SimVisionSystem.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (C) 2020 Photon Vision.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.photonvision;
+
+import java.util.LinkedList;
+
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Transform2d;
+import edu.wpi.first.wpiutil.math.Pair;
+
+public class SimVisionSystem {
+    SimPhotonCamera cam;
+
+    double camDiagFOV_deg;
+    double camHorizFOV_deg;
+    double camVertFOV_deg;
+    double cameraHeightOffGround_m;
+    double maxLEDRange_m;
+    double camPitch_deg;
+    Pair<Integer, Integer> cameraRes_px;
+    double minTargetArea;
+    Transform2d robotToCamera;
+
+    LinkedList<SimVisionTarget> tgtList;
+
+    /**
+     * Create a simulated vision system involving a camera and coprocessor mounted on a mobile robot running Photonvision, detecting one or more targets scattered around the field.
+     * This assumes a fairly simple and distortionless pinhole camera model.
+     * @param camName Name of the photonvision camera to create. Align it with the settings you use in the PhotonVision GUI.
+     * @param camDiagFOV_deg Diagonal Field of View of the camera used. Align it with the manufacturer specifications, and/or whatever is configured in the PhotonVision Setting page.
+     * @param camPitch_deg pitch of the camera's view axis back from horizontal. Make this the same as whatever is configured in the PhotonVision Setting page.
+     * @param robotToCamera Pose Transform to move from the robot's position to the camera's mount position
+     * @param cameraHeightOffGround_m Height of the camera off the ground in meters
+     * @param maxLEDRange_m Maximum distance at which your camera can illuminate the target and make it visible. Set to 9000 or more if your vision system does not rely on LED's.
+     * @param cameraRes_px Pair of integers representing the Width and Height (respectively) of your camera's image sensor in pixels
+     * @param minTargetArea Minimum area that that the target should be before it's recognized as a target by the camera. Match this with your contour filtering settings in the PhotonVision GUI.
+     */
+    public SimVisionSystem(String camName, double camDiagFOV_deg, double camPitch_deg, Transform2d robotToCamera, double cameraHeightOffGround_m, double maxLEDRange_m, Pair<Integer, Integer> cameraRes_px, double minTargetArea){
```

Comment:
10-4, I'll confirm whatever they have and follow suit. the "Pair" was an arbitrary invention and choice on my end.

Reply:
FWIW I'd attempted to avoid using openCV types as part of public API's, but that's really the only driving reason I had not to do `Pair`. "Match wpilib" is still my thought for best strategy.